### ISSUE TITLE
When are the weights of layers initialized and how to reinitialize them?

### ISSUE THREAD

**Issue by cdeepakroy** [keras-team/keras#9362]
It is not clear at what point are the weights of the layers initialized?

-- Are they initialized when the Model() object is created? or
-- Are they initialized when the model is compiled by calling `model.compile()`?

I am hoping a different random seed is used each time they are initialized. Is that so?

How can i re-initialize the weights of the layers (only if `layer.trainable is True`) with a different random seed?
